components/calendar/src/astronomy.rs

```diff
@@ -1368,37 +1373,30 @@ impl Astronomical {
 
     /// The parallax of the moon, meaning the difference in angle of the direction of the moon
     /// as measured from a given location and from the center of the Earth, in degrees.
-    ///
     /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz,
     /// originally from _Astronomical Algorithms_ by Jean Meeus, 2nd edn., 1998.
     /// Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/9afc1f3/calendar.l#L4619-L4628
-    #[allow(dead_code)] // TODO: Remove dead code tag after use
-    pub(crate) fn lunar_parallax(moment: Moment, location: Location) -> f64 {
-        let geo = Self::lunar_altitude(moment, location);
+    pub(crate) fn lunar_parallax(lunar_altitude_val: f64, moment: Moment) -> f64 {
         let cap_delta = Self::lunar_distance(moment);
         let alt = 6378140.0 / cap_delta;
-        let arg = alt * libm::cos(geo.to_radians());
+        let arg = alt * libm::cos(lunar_altitude_val.to_radians());
         arcsin_degrees(arg)
     }
 
-    /// Topocentric altitude of the moon, meaning the celestial altitude of the moon with the given
-    /// location as the origin, at a given moment, ignoring refraction.
-    ///
+    /// Topocentric altitude of the moon.
     /// Based on functions from _Calendrical Calculations_ by Reingold & Dershowitz.
     /// Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/9afc1f3/calendar.l#L4630-L4636
-    #[allow(dead_code)] // TODO: Remove dead code tag after use
-    fn topocentric_lunar_altitude(moment: Moment, location: Location) -> f64 {
-        Self::lunar_altitude(moment, location) - Self::lunar_parallax(moment, location)
+    fn topocentric_lunar_altitude(lunar_altitude_val: f64, lunar_parallax_val: f64) -> f64 {
+        lunar_altitude_val - lunar_parallax_val
```

Comment:
question: do we need a separate function for this?

Reply:
We do not, but I believe it is a cleaner solution, plus it comes from the book and we can reference it.

Reply:
Suggestion: I see only one call site for this function. It's not clear that optimizing this one gives us a perf boost. Maybe revert the changes here to be more consistent with Reingold when in doubt.